src/modm/platform/uart/stm32/uart_hal.hpp.in

```diff
@@ -44,6 +45,10 @@ class {{ hal }} : public UartBase
 {
 public:
 	static constexpr bool isExtended = {{ extended_driver | lower }};
+    static const Peripheral UartPeripheral = Peripheral::{{name}};
+    static inline std::function<bool(void)> RxCallback;
+    static inline std::function<bool(void)> TxCallback;
+    static inline std::function<void(bool)> FinalCallback;
```

Comment:
```suggestion
    static inline modm::inline_function<void(bool)> FinalCallback;
```
`std::function` uses the heap.

Reply:
Good point, fixed that.

Reply:
@salkinium `modm::inline_function` currently has a default capacity of only `sizeof(void*)`. This wouldn't let you store any lambda with a capture. If we start using it in interfaces across the whole library we should rethink what default to choose. One pointer seems too restrictive targeting anything bigger than an ATTiny.